### DISCUSSION TITLE
There has anyway to override the Chrome's / OS's shortcuts ?

### DISCUSSION THREAD

**Discussion by howel52** [eclipse-theia/theia#8118] [General] [answered]
### Feature Description:
Some shortcuts are often used like `cmd + w`, `cmd + t` when we use the VSCode(Electron App).

But in the web environment,  there has some conflicts  between `Web Editor` And `Browser`

Has anyway to resolve this problem ?

Maybe we need some feature to proxy user's keyboard event  in the web environment？  Or in some other ways.

**Comment by akosyakov** [comment, marked as answer]
Generally it is impossible as for now browsers don't allow to override certain shortcuts. There is a special browser window modes which allow it, i.e. a pop-up window of full-screen mode. But that's it. One possible solution would be to provide Electron Theia but which talks to remote Theia backend.

**Comment by howel52** [comment]
I have an idea and tried it a few days ago, and solved this problem.

We developed a `Chrome Extension` to block some keyboardEvents like `cmd + w`.

And then we emit some keyboardEvents what we needed in the `Extension` when user have been focus on our web application.   : )

**Reply by akosyakov** [reply]
Could you share the extension?

**Reply by howel52** [reply]
Our primary goal is to solve the problems for our company's customers, so there will be some private code in our extension.  

But we will consider deleting our private code in the next stage and make a general solution with open source.

Maybe next month : )
